torchsde/_core/base_sde.py

```diff
@@ -91,10 +91,10 @@ def gdg_prod_default(self, t, y, v):
         return vg_dg_vjp
 
     def gdg_prod_diagonal(self, t, y, v):
-        requires_grad = torch.is_grad_enabled()
+        requires_grad = torch.is_grad_enabled() and (t.requires_grad or y.requires_grad or v.requires_grad)
```

Comment:
Do we have an application/use-case where we need to call torch.autograd.grad where t is the input?

Reply:
See below.